addons/sale_stock/models/sale_order_line.py

```diff
@@ -14,7 +14,7 @@ class SaleOrderLine(models.Model):
     _inherit = 'sale.order.line'
 
     qty_delivered_method = fields.Selection(selection_add=[('stock_move', 'Stock Moves')])
-    route_id = fields.Many2one('stock.route', string='Route', domain=[('sale_selectable', '=', True)], ondelete='restrict', check_company=True)
+    route_id = fields.Many2one('stock.route', string='Route', domain=[('sale_selectable', '=', True)], ondelete='restrict')
```

Comment:
why? are we sure we don't want to filter out routes from other companies anymore?

Reply:
The idea is to delivery from another company. Currently, you can do it with a make to order route and the dropship route. You also have to use the other company as the vendor.
The benefits of the current solution is the valuation because it will create PO/SO/Invoice and create the correct account entries for both company.
Removing route, allow you to deliver directly from another company and give more flexibility because you can choose between the companies. It will also create fewer documents (only the delivery) that are not necessary, needed if you don't do your accounting in odoo.